src/mixins/ProjectorMixin.ts

```diff
@@ -0,0 +1,207 @@
+import global from '@dojo/core/global';
+import Promise from '@dojo/shim/Promise';
+import { createProjector as createMaquetteProjector, Projector as MaquetteProjector } from 'maquette';
+import { EventTargettedObject, Handle } from '@dojo/interfaces/core';
+import { WidgetConstructor, WidgetProperties } from './../WidgetBase';
+import { Constructor } from './../interfaces';
+
+/**
+ * Represents the state of the projector
+ */
+export enum ProjectorState {
+	Attached = 1,
+	Detached
+};
+
+/**
+ * Attach type for the projector
+ */
+export enum AttachType {
+	Append = 1,
+	Merge = 2,
+	Replace = 3
+};
+
+export interface AttachOptions {
+
+	/**
+	 * If `'append'` it will append to the root. If `'merge'` it will merge with the root. If `'replace'` it will
+	 * replace the root.
+	 */
+	type: AttachType;
+}
+
+export interface ProjectorProperties extends WidgetProperties {
+
+	root?: Element;
+
+	cssTransitions?: boolean;
+}
+
+export interface Projector {
+
+	/**
+	 * Append the projector to the root.
+	 */
+	append(): Promise<Handle>;
+
+	/**
+	 * Merge the projector onto the root.
+	 */
+	merge(): Promise<Handle>;
+
+	/**
+	 * Replace the root with the projector node.
+	 */
+	replace(): Promise<Handle>;
+
+	/**
+	 * Root element to attach the projector
+	 */
+	root: Element;
+
+	/**
+	 * The status of the projector
+	 */
+	readonly projectorState: ProjectorState;
+}
+
+export function ProjectorMixin<T extends WidgetConstructor>(base: T): T & Constructor<Projector> {
```

Comment:
I have done this for all classes, it's a topic for discussion i guess but I you'll rarely just use the default.